go-kit/metrics/provider/otel/types.go

```diff
@@ -45,10 +45,18 @@ func (p *Provider) NewCounter(name string) metrics.Counter {
 }
 
 func (p *Provider) newCounter(name string, labelValues ...string) metrics.Counter {
+	k := keyName(name, labelValues...)
+
+	p.mu.RLock()
+	c, ok := p.counters[k]
```

Comment:
nit: Not sure if this can be moved to provider.go and get reused.

Reply:
And by "this" you mean the functions in this file?